Mathlib/Analysis/InnerProductSpace/Projection.lean

```diff
@@ -586,6 +587,33 @@ variable (K)
 theorem orthogonalProjection_norm_le : ‖K.orthogonalProjection‖ ≤ 1 :=
   LinearMap.mkContinuous_norm_le _ (by norm_num) _
 
+/-- The orthogonal projection onto a closed subspace is norm non-increasing. -/
+theorem norm_orthogonalProjection_apply_le (v : E) :
+    ‖orthogonalProjection K v‖ ≤ ‖v‖ := by calc
+  ‖orthogonalProjection K v‖ ≤ ‖orthogonalProjection K‖ * ‖v‖ := K.orthogonalProjection.le_opNorm _
+  _ ≤ 1 * ‖v‖ := by gcongr; exact orthogonalProjection_norm_le K
+  _ = _ := by simp
+
+/-- The orthogonal projection onto a closed subspace is a `1`-Lipschitz map. -/
+theorem lipschitzWith_orthogonalProjection :
+    LipschitzWith 1 (orthogonalProjection K) :=
+  ContinuousLinearMap.lipschitzWith_of_opNorm_le (orthogonalProjection_norm_le K)
+
+/-- The operator norm of the orthogonal projection onto a nontrivial subspace is `1`. -/
+theorem norm_orthogonalProjection (hK : K ≠ ⊥) :
+    ‖K.orthogonalProjection‖ = 1 := by
+  refine le_antisymm K.orthogonalProjection_norm_le ?_
+  obtain ⟨x, hxK, hx_ne_zero⟩ := Submodule.exists_mem_ne_zero_of_ne_bot hK
+  set x' : E := (‖x‖⁻¹ : 𝕜) • x
+  have hx'_mem : x' ∈ K := K.smul_mem _ hxK
+  have hx'_norm : ‖x'‖ = 1 := norm_smul_inv_norm hx_ne_zero
+  have h_proj : K.orthogonalProjection x' = x' := orthogonalProjection_eq_self_iff.mpr hx'_mem
+  calc
+    1 = ‖x'‖ := symm hx'_norm
+    _ = ‖K.orthogonalProjection x'‖ := by simp only [AddSubgroupClass.coe_norm, h_proj]
+    _ ≤ ‖orthogonalProjection K‖ * ‖x'‖ := ContinuousLinearMap.le_opNorm _ _
+    _ = ‖orthogonalProjection K‖ := by rw [hx'_norm]; simp
```

Comment:
No need to use a unit vector, the lemma you want here is `ContinuousLinearMap.ratio_le_opNorm` (or `unit_le_opNorm` if you really want an element of the unit ball).
```suggestion
  simpa [K.norm_orthogonalProjection_apply hxK, norm_eq_zero, hx_ne_zero]
    using K.orthogonalProjection.ratio_le_opNorm x
```